Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject/DownloadObject.cs

```diff
@@ -11,216 +11,179 @@
 using System.Text.RegularExpressions;
 using System.Linq;
 using Frends.AmazonS3.DownloadObject.Definitions;
+using System.Reflection;
+using System.Runtime.Loader;
 
 namespace Frends.AmazonS3.DownloadObject;
 
 /// <summary>
-/// Amazon S3 task.
+/// Amazon S3 Task.
 /// </summary>
 public class AmazonS3
 {
+
+    /// For mem cleanup.
+    static AmazonS3()
+    {
+        var currentAssembly = Assembly.GetExecutingAssembly();
+        var currentContext = AssemblyLoadContext.GetLoadContext(currentAssembly);
+        if (currentContext != null)
+            currentContext.Unloading += OnPluginUnloadingRequested;
+    }
+    
+    private static readonly HttpClient Client = new();
+
     /// <summary>
     /// Download object(s) from AWS S3.
     /// [Documentation](https://tasks.frends.com/tasks#frends-tasks/Frends.AmazonS3.DownloadObject)
     /// </summary>
     /// <param name="input">Input parameters</param>
     /// <param name="cancellationToken">Token generated by Frends to stop this task.</param>
-    /// <returns>List { string ObjectData, string ObjectName, string FullPath }</returns>
+    /// <returns>List { string ObjectName, string FullPath, string Overwritten, bool SourceDeleted, string Info }</returns>
     public static async Task<Result> DownloadObject([PropertyTab] Input input, CancellationToken cancellationToken)
     {
+        var result = new List<SingleResultObject>();
+
         if (string.IsNullOrWhiteSpace(input.DestinationDirectory)) throw new Exception($"Destination required.");
 
-        switch (input.AuthenticationMethod)
+        if (input.AuthenticationMethod.Equals(AuthenticationMethod.AWSCredentials))
```

Comment:
You can use `input.AuthenticationMethod is AuthenticationMethod.AWSCredentials` instead of calling `object.Equals`.